marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/models/generateModelConfig.sjs

```diff
@@ -0,0 +1,62 @@
+/*
+  Copyright (c) 2020 MarkLogic Corporation
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+'use strict';
+
+const consts = require("/data-hub/5/impl/consts.sjs");
+const config = require("/com.marklogic.hub/config.sjs");
+const hent = require("/data-hub/5/impl/hub-entities.xqy");
+
+
+const result = {};
+const protectedPaths = [];
+const queryRolesets = [];
+const index = {
+  "database-names": [],
+  "index-config": {}
+};
+const options = {};
+
+const entityModels = cts.search(cts.collectionQuery(consts.ENTITY_MODEL_COLLECTION)).toArray();
+if (entityModels.length > 0) {
+  // Add PII files
+  const securityConfig = hent.dumpPii(entityModels).toObject();
+  const protectedPathsExist = securityConfig.config && securityConfig.config["protected-path"] && securityConfig.config["protected-path"].length > 0;
+  if (protectedPathsExist) {
+    securityConfig.config["protected-path"].forEach(path => {
+      protectedPaths.push(xdmp.toJSON(path));
+    });
+    if (securityConfig.config["query-roleset"]) {
+      queryRolesets.push(xdmp.toJSON(securityConfig.config["query-roleset"]));
+    }
+  }
+
+  // Add search options
+  options["default"] = hent.dumpSearchOptions(entityModels, false);
+  options["explorer"] = hent.dumpSearchOptions(entityModels, true);
+
+  // Add indexes
+  const dbProps = hent.dumpIndexes(entityModels);
+  index["index-config"] = dbProps.toObject();
+  index["database-names"].push(config.STAGINGDATABASE);
```

Comment:
I don't think you should add the database-names here - it's assumed that the client knows what those are, which is the case here. If you were doing it here, you arguably should do it for search-options as well. But I don't think it's necessary - the middle tier has a HubClient, so it's good to go.

Reply:
Updated in here b6777af to use the middle tier for that database names.